components/movementsensor/gpsnmea/serial.go

```diff
@@ -206,7 +212,21 @@ func (g *SerialNMEAMovementSensor) Orientation(ctx context.Context, extra map[st
 func (g *SerialNMEAMovementSensor) CompassHeading(ctx context.Context, extra map[string]interface{}) (float64, error) {
 	g.mu.RLock()
 	defer g.mu.RUnlock()
-	return 0, movementsensor.ErrMethodUnimplementedCompassHeading
+	currentHeading := g.data.CompassHeading
+	lastHeading := g.lastcompassheading.GetLastCompassHeading()
+
+	if !math.IsNaN(currentHeading) && !math.IsNaN(lastHeading) && currentHeading != lastHeading {
+		g.lastcompassheading.SetLastCompassHeading(currentHeading)
+	}
+
+	if math.IsNaN(lastHeading) && !math.IsNaN(currentHeading) {
```

Comment:
Same suggestion here about finding ways to make the if statements simpler and less redundant.